test/e2e/vertical_scaling_test.go

```diff
@@ -0,0 +1,170 @@
+	machineToClone.Name = fmt.Sprintf("%s-clone", machineToClone.Name)
+	machineToClone.Spec.ProviderID = nil
+	machineToClone.ResourceVersion = ""
+
+	clonedMachine, err := machineClient.Create(context.TODO(), machineToClone, metav1.CreateOptions{})
+	require.NoError(t, err)
+
+	t.Logf("Created a new master machine/node %q", clonedMachine.Name)
+	return clonedMachine.Name
+}
+
+func ensureMasterMachineRunning(t testing.TB, machineName string, machineClient machinev1beta1client.MachineInterface) {
+	waitPollInterval := 15 * time.Second
+	waitPollTimeout := 5 * time.Minute
+	t.Logf("Waiting up to %s for %q machine to be in the Running state", waitPollTimeout.String(), machineName)
+
+	if err := wait.Poll(waitPollInterval, waitPollTimeout, func() (bool, error) {
+		machine, err := machineClient.Get(context.TODO(), machineName, metav1.GetOptions{})
+		if err != nil {
+			return false, err
+		}
+		machinePhase := pointer.StringDeref(machine.Status.Phase, "Unknown")
+		t.Logf("%q machine is in %q state", machineName, machinePhase)
+		if machinePhase == "Running" {
+			return true, nil
+		}
+		return false, nil
+	}); err != nil {
+		newErr := fmt.Errorf("failed to check if %q is Running state, err: %v", machineName, err)
+		require.NoError(t, newErr)
+	}
+}
+
+// asserts the cluster has only 3 master machines and all are in the Running state
+func ensureCluster(t testing.TB, machineClient machinev1beta1client.MachineInterface) {
+	machineList, err := machineClient.List(context.TODO(), metav1.ListOptions{LabelSelector: masterMachineLabelSelector})
+	require.NoError(t, err)
+
+	if len(machineList.Items) != 3 {
+		machineNames := []string{}
+		for _, machine := range machineList.Items {
+			machineNames = append(machineNames, machine.Name)
+		}
+		t.Fatalf("incorrect cluster size, expected exactly 3 master machines, got: %v, found: %v", len(machineList.Items), machineNames)
+	}
+
+	for _, machine := range machineList.Items {
+		machinePhase := pointer.StringDeref(machine.Status.Phase, "")
+		if machinePhase != "Running" {
+			t.Fatalf("%q machine is in unexpected %q state", machine.Name, machinePhase)
+		}
+	}
+}
+
+// ensureMembersCount simply counts the current etcd members, it doesn't evaluate health conditions or any other attributes (i.e. name) of individual members
+func ensureMembersCount(t testing.TB, cs *framework.ClientSet, expectedMembersCount int) {
+	etcdPods, err := cs.CoreV1Interface.Pods("").List(context.TODO(), metav1.ListOptions{LabelSelector: "k8s-app=etcd"})
+	require.NoError(t, err)
+	etcdPodName := etcdPods.Items[0].Name
+
+	waitPollInterval := 15 * time.Second
+	waitPollTimeout := 5 * time.Minute
+	t.Logf("Waiting up to %s for a new etcd member to join the cluster", waitPollTimeout.String())
+
+	if err := wait.Poll(waitPollInterval, waitPollTimeout, func() (bool, error) {
+		ocCmd := strings.Split(fmt.Sprintf("rsh -c etcdctl -n openshift-etcd %s %s", etcdPodName, "etcdctl member list -w=json"), " ")
```

Comment:
I am using `oc` for getting the member list, in the future, we could consume etcd API. I think it is a good starting point.